File: src/main/java/com/upokecenter/text/encoders/AppResources.java
package com.upokecenter.text.encoders;

import java.util.*;

  final class AppResources {

    private ResourceBundle mgr;

    public AppResources(String name) {
      this.mgr = ResourceBundle.getBundle(name);
    }

    public short[] GetShortArray(String name) {
      String str = this.mgr.getString(name);
      if(str==null || (str.length()&3) != 0){
         return null;
      }
      short[] ret=new short[str.length()>>2];
      int k=0;
      for(int i=0;i<str.length();i+=4){
        int c=0;
        int shift=12;
        for(int j=0;j<4;j++,shift-=4){
         int d=str.charAt(i+j);
         if(d>='0' && d<='9'){
           d=(int)(d-'0');
         } else if(d>='a' && d<='f'){
           d=(int)(d-'a')+10;
         } else if(d>='A' && d<='F'){
           d=(int)(d-'A')+10;
         } else return null;
         c|=(d<<shift);
        }
        ret[k++]=(short)c;
      }
      return ret;
    }

    public int[] GetIntArray(String name) {
      String str = this.mgr.getString(name);
      if(str==null || (str.length()&7) != 0){
         return null;
      }
      int[] ret=new int[str.length()>>3];
      int k=0;
      for(int i=0;i<str.length();i+=8){
        int c=0;
        int shift=28;
        for(int j=0;j<8;j++,shift-=4){
         int d=str.charAt(i+j);
         if(d>='0' && d<='9'){
           d=(int)(d-'0');
         } else if(d>='a' && d<='f'){
           d=(int)(d-'a')+10;
         } else if(d>='A' && d<='F'){
           d=(int)(d-'A')+10;
         } else return null;
         c|=(d<<shift);
        }
        ret[k++]=c;
      }
      return ret;
    }

    public String GetString(String name) {
      return mgr.getString(name);
    }
  }

File: Encoding/PeterO/Text/Encoders/AppResources.cs
using System.Resources;

#pragma warning disable CA1801

namespace PeterO.Text.Encoders {
  internal sealed class AppResources {
    private readonly ResourceManager mgr;

    public AppResources(string name) {
      this.mgr = new ResourceManager(this.GetType());
    }

    public short[] GetShortArray(string name) {
      string str = this.mgr.GetString(name,
          System.Globalization.CultureInfo.InvariantCulture);
      if (str == null || (str.Length & 3) != 0) {
        return null;
      }
      var ret = new short[str.Length >> 2];
      var k = 0;
      for (var i = 0; i < str.Length; i += 4) {
        var c = 0;
        var shift = 12;
        for (var j = 0; j < 4; j++, shift -= 4) {
          int d = str[i + j];
          if (d >= '0' && d <= '9') {
            d = (int)(d - '0');
          } else if (d >= 'a' && d <= 'f') {
            d = (int)(d - 'a') + 10;
          } else if (d >= 'A' && d <= 'F') {
            d = (int)(d - 'A') + 10;
          } else {
            return null;
          }
          c |= d << shift;
        }
        ret[k++] = unchecked((short)c);
      }
      return ret;
    }

    public int[] GetIntArray(string name) {
      string str = this.mgr.GetString(name,
          System.Globalization.CultureInfo.InvariantCulture);
      if (str == null || (str.Length & 7) != 0) {
        return null;
      }
      var ret = new int[str.Length >> 3];
      var k = 0;
      for (var i = 0; i < str.Length; i += 8) {
        var c = 0;
        var shift = 28;
        for (var j = 0; j < 8; j++, shift -= 4) {
          int d = str[i + j];
          if (d >= '0' && d <= '9') {
            d = (int)(d - '0');
          } else if (d >= 'a' && d <= 'f') {
            d = (int)(d - 'a') + 10;
          } else if (d >= 'A' && d <= 'F') {
            d = (int)(d - 'A') + 10;
          } else {
            return null;
          }
          c |= d << shift;
        }
        ret[k++] = c;
      }
      return ret;
    }

    public string GetString(string name) {
      return this.mgr.GetString(name,
          System.Globalization.CultureInfo.InvariantCulture);
    }
  }
}
